### PR TITLE
Add debug ability to the telemetry operator

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- The telemetry operator would look for override configmap at the start of reconciliation and would:
    * pause if the the pause flag is set in either logging or tracing config.
    * Change log level of controller runtime on fly.
- To keep code simple a lazy implementation has been implemented, where one needs to trigger reconciliation after adding the configmap to the kyma-system namespace
- Fast integration tests are implemented [here](https://github.com/kyma-project/kyma/pull/16616)

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/16435

### CODE REVIEW

File: components/telemetry-operator/controller/logpipeline/controller.go [hunk 1]
```diff
@@ -54,6 +57,11 @@ type DaemonSetAnnotator interface {
 	SetAnnotation(ctx context.Context, name types.NamespacedName, key, value string) error
 }
 
+type GlobalConfigHandler interface {
```
Comment (chrkl) [inline components/telemetry-operator/controller/logpipeline/controller.go:60]: Let's move this interface to the overrides package (same for tracepipeline/controller.go)

File: components/telemetry-operator/internal/kubernetes/configmap.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ConfigmapProber struct {
+	client.Client
+}
+
+const overrideConfigFileName = "override-config"
+
+func (cmp *ConfigmapProber) ReadConfigMapIfPresent(ctx context.Context, name types.NamespacedName) (string, error) {
```
Comment (chrkl) [inline components/telemetry-operator/internal/kubernetes/configmap.go:19]: The behaviour of returning an empty string if the CM does not exist might be not expected for that method name. Maybe let's name it `ReadConfigMapOrEmpty`

File: components/telemetry-operator/internal/overrides/overrides.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	if len(config) == 0 {
+		return overrideConfig, nil
+	}
+
+	err = yaml.Unmarshal([]byte(config), &overrideConfig)
+	if err != nil {
+		return overrideConfig, err
+	}
+
+	log.V(1).Info(fmt.Sprintf("Override Config is: %+v", overrideConfig))
```
Comment (chrkl) [inline components/telemetry-operator/internal/overrides/overrides.go:70]: Follow our [conventions](https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md#log-structure) for log messages. Maybe better have a message like `Using override config ...`

File: components/telemetry-operator/internal/configureLogger/configurelogger.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package configurelogger
```
Comment (chrkl) [inline components/telemetry-operator/internal/configureLogger/configurelogger.go:1]: I miss the `Good package names are short and clear.` aspect from [this](https://go.dev/blog/package-names) article. How about moving this to the `logger` package?

File: components/telemetry-operator/controller/logparser/controller.go [hunk 5]
```diff
@@ -46,22 +49,29 @@ type DaemonSetAnnotator interface {
 	SetAnnotation(ctx context.Context, name types.NamespacedName, key, value string) error
 }
 
+type GlobalConfigHandler interface {
```
Comment (chrkl) [inline components/telemetry-operator/controller/logparser/controller.go:52]: Can be removed here, right?

File: components/telemetry-operator/internal/configureLogger/configurelogger.go (hunk 4 above)
Comment (rakesh-garimella) [reply components/telemetry-operator/internal/configureLogger/configurelogger.go]: fixed

**Review by chrkl — APPROVED** [review]

**Pull request merged**